### ISSUE TITLE
displaying transcript isoform-specific annotations

### ISSUE THREAD

**Issue by mah11** [pombase/website#1613]
Now that we have a few real live "alternative" transcripts annotated, we ought to think about what the web (gene, publication, etc.) pages should do with annotations to them. 

There are already examples where the transcripts have different annotated:
- product descriptions (e.g. wtf4)
- GO annotations (e.g. wtf13)
- expression (e.g. tco1)
- phenotype extensions (tco1 in sre1 phenotype)

At this point I'm mainly opening this for discussion and so it won't fall off the radar; I don't have well-formed ideas about what we should actually do. them. I've opened a [Canto ticket](https://github.com/pombase/canto/issues/2351) for making isoform-specific annotations too. Will we also need a Chado ticket?

**Comment by mah11** [comment]
One aspect of the current behavior that we won't want long-term: it appears to be arbitrarily taking the product description for the longest transcript, and not using the other(s) at all (e.g. wtf4, tco1).

**Comment by ValWood** [comment]
OK let's kick off with the product

We could duplicate the  header section for each transcript since the information for each will be different:


<img width="958" alt="Screenshot 2020-12-15 at 20 53 34" src="https://user-images.githubusercontent.com/7359272/102271933-e55ff180-3f17-11eb-85f1-e2e0bcbd36c5.png">


I don't have any good ideas for the annotation.
At the moment I'm thinking that  we default to  one and toggle over to an identical display for the other isoforms.

**Comment by ValWood** [comment]
Here is my first idea as a starting point:

<img width="462" alt="Screenshot 2020-12-15 at 21 03 34" src="https://user-images.githubusercontent.com/7359272/102272854-5653d900-3f19-11eb-8641-497182488879.png">

**Comment by kimrutherford** [comment]
> We could duplicate the header section for each transcript since the information for each will be different:

I think it would confuse things if we used the transcript Systematic ID in the header instead of the gene ID.

Perhaps we could split the header into two, one for gene stuff and a for information that can change per transcript:
  - transcript ID
  - location
  - weight

> Here is my first idea as a starting point:

Do we have transcript specific annotation?  Maybe a filter on the annotation table would be OK?

**Comment by mah11** [comment]
> Do we have transcript specific annotation?

A little bit, yes, mostly for some of the wtfs so far.

> Maybe a filter on the annotation table would be OK?

I'd be willing to give it a try.

**Comment by kimrutherford** [comment]
> Perhaps we could split the header into two, one for gene stuff and a for information that can change per transcript:

I've made a mock-up which I think would be OK for a gene with couple of transcripts but we'd need something cleverer for genes with lots of transcripts (will those be a thing?).  Perhaps for many-transcript genes, a table would make sense with columns like "ID", "translation start", "UTR end" and "product size"?

Anyway:

![multi-transcript-details-1](https://user-images.githubusercontent.com/90474/112103526-c2c42f80-8c0e-11eb-9956-5be62e11bdad.png)

**Comment by kimrutherford** [comment]
>  Perhaps for many-transcript genes, a table would make sense

Mock-up:
![multi-transcript-details-2](https://user-images.githubusercontent.com/90474/112116026-b9db5a00-8c1e-11eb-9050-90d7bbe2daa4.png)

**Comment by mah11** [comment]
> I've made a mock-up which I think would be OK for a gene with couple of transcripts 

That looks good.

> genes with lots of transcripts (will those be a thing?)

I don't know. The most I've tried curating so far is 3, iirc.

> ... for many-transcript genes, a table ... mock-up

That looks good, but it also looks partly redundant with the Transcript section. Do we need a re-think about which details go in which section?

Also, would there be a table in every gene page header, or only for genes with more than X transcripts (for whatever value of X)? In the latter case, will inconsistency bother users?

Am I overthinking this?

**Comment by ValWood** [comment]
- I think above 4 will be very rare. We won't often annotate multiple transcripts , but 2 will be most common i.e. vegetative/meiotic  which are often very different lengths.

- I think the redundancy is OK, it is only start and end that is repeated and that is really summary info. The 'transcript section' is more about "transcript structure. 
- I would be happy for this to only appear if multiple transcripts are annotated

- we would still need to decide  how to represent products (and link to their respective transcript IDs)

**Comment by kimrutherford** [comment]
Yep, I was thinking we'd only show a transcript table (or whatever) if there are multiple transcripts.  Although in the single transcript case we could put "This gene has one curated transcript" where the table would be, just for clarity.

> I think the redundancy is OK, it is only start and end that is repeated 

I don't understand this bit.  Can't each transcript potentially have different starts and ends?

**Comment by ValWood** [comment]
> I think the redundancy is OK, it is only start and end that is repeated

not clear sorry. I was saying that there isn't a total overlap between the "alternative transcript table" in the summary, and the transcript section. It's slightly redundant, but the only data that is in both sections is transcript  coordinates.

**Comment by kimrutherford** [comment]
Ah, right.  Got it now.

Another option is to have less information about the transcripts at the top of the page and instead have a link to the Transcripts section:
```
Note: this gene has 2 annotated transcripts.  [Details...]
```

then we could collect all the transcript specific stats in one place.

**Comment by mah11** [comment]
> have less information about the transcripts at the top of the page and instead have a link to the Transcripts section

ooh, I like that idea - it would make the top of the page a bit more streamlined

**Comment by ValWood** [comment]
+1

**Comment by kimrutherford** [comment]
> instead have a link to the Transcripts section

I'll do that soon as it will probably help even without any other changes.

**Comment by kimrutherford** [comment]
> instead have a link to the Transcripts section

I've added that but only for multi-transcript genes for now.

![multi-transcript-details-3](https://user-images.githubusercontent.com/90474/112400788-f6699b80-8d6d-11eb-876e-50e4eac21399.png)

**Comment by ValWood** [comment]
I'm not sure but I don't think we necessarily want to display the longest by default. The default transcript and product used should be -1.

So for example tco1
https://www.pombase.org/gene/SPAC17G6.02c
would show the translated transcript.

Perhaps as long as we know there are multiple transcripts the length can be confined to the transcript section?

**Comment by ValWood** [comment]
> I've added that but only for multi-transcript genes for now.

This is good. Should we do:

Note: This gene has 2 transcripts [details] and 2 translations

to distinguish which have different products?

**Comment by kimrutherford** [comment]
> Should we do:
> Note: This gene has 2 transcripts [details] and 2 translations
> to distinguish which have different products?

Yep, that makes sense.  Although at the moment there are always the same number of transcripts and translations.

**Comment by ValWood** [comment]
> Although at the moment there are always the same number of transcripts and translations.

Not for tco1 (although there is currently a CDS, there shouldn't be). But that could be resolved by
https://github.com/pombase/pombase-chado/issues/822

**Comment by kimrutherford** [comment]
We decided to add the product size to the "Transcript details" section.

Also if there are multiple transcripts we will say "Product size (of transcript 1)   123aa, 321 kDa" in the summary at the top of the page.

**Comment by kimrutherford** [comment]
> We decided to add the product size to the "Transcript details" section.
> Also if there are multiple transcripts we will say "Product size (of transcript 1) 123aa, 321 kDa" in the summary at the top of the page.

I've done that and added per-transcript product sizes in the "Transcript details" section.

I suspect it will need tweaking when we have multi-transcript genes where only some transcripts are protein coding.

**Comment by kimrutherford** [comment]
Looking through this issue it seems like there's still quite a bit to do here.  In particular we need to decide how to display transcript specific annotations.  I can't think of a nice way to do that.

I think a table for each transcript and each annotation type would be a mess.  Especially since I'm guessing that a lot of annotation won't be transcript specific so we'd need a table for each transcript and also one for annotations to the gene where the transcript isn't annotated (shared annotation).

We could add yet another filter on the annotation tables, but that seems quite hidden?

Or have an extra column in the annotation table show which transcript(s) or protein(s) the annotation applies to?  That also seems a bit hidden and makes the display busier.

**Comment by kimrutherford** [comment]
> Or have an extra column in the annotation table show which transcript(s) or protein(s) the annotation applies to? That also seems a bit hidden and makes the display busier.

Midori and I had a chat about this about and we think this plan might be OK.  The GO annotation tables aren't that busy and we would only show the extra column when it was needed.

It makes sense to have a new filter on the table too.

Other things to think about, which will need new issues at some point:

  - how should we record the isoform when making the annotation in Canto?  Would an extra (optional) extension be OK?
  - need to fix the Chado loading code to handle isoform specific annotation
     - the annotation is attached to the transcripts in Chado, but currently everything from Canto is attached to the first transcript
     - the contig loading works OK and it's possible to attach annotation to any transcript
  - need to handle GO annotation that applies to all transcripts
     - the straight forward way (in Chado) is to annotate both transcripts in that case
     - the website code will need to change to cope with this - we don't want to show the same annotation twice
  - the GO filtering code in the loader needs to cope with annotation on transcripts
  - the code that exports transcript and protein sequences as fasta needs to change

**Comment by mah11** [comment]
Thanks - that covers everything I remember.

> how should we record the isoform when making the annotation in Canto? Would an extra (optional) extension be OK?

eh, probably

**Comment by kimrutherford** [comment]
One other thing we talked about was the "Protein features" section of the gene pages, which mostly comes from Pfam and InterPro.

I looked up wtf4 in Pfam and it looks like they have only one isoform: http://pfam.xfam.org/protein/Q9P3V0

InterPro have both:
  - https://www.ebi.ac.uk/interpro/protein/reviewed/Q9P3V0/?isoform=Q9P3V0-1
  - https://www.ebi.ac.uk/interpro/protein/reviewed/Q9P3V0/?isoform=Q9P3V0-2

Some of the "Protein features" section is transcript specific so we might need a selector like:

![multi-transcript-prot-feat-1](https://user-images.githubusercontent.com/90474/120631911-82114e00-c4bc-11eb-950b-31c5be3340a4.png)

**Comment by ValWood** [comment]
I like this plan.

This ticket isn't very urgent in comparison to other things because there are so few annotated. 

prp10 is a useful example of a gene which has multiple transcripts but should collapse because it only has one protein product:
<img width="1063" alt="alternative transcripts" src="https://user-images.githubusercontent.com/7359272/125203685-5e86b380-e271-11eb-8868-1937c9911f29.png">
